### PR TITLE
Finalize async runtime structure

### PR DESCRIPTION
**Pull request by wusyong**

### CODE REVIEW

File: src/node/supervisor.rs [hunk 1]
```diff
@@ -80,16 +79,16 @@ impl Supervisor {
                 Event::GetShardsNum => {
                     // TODO connect to scylla-shard-zero and get_cql_opt to finally get the shards_num
                     // for testing purposes, we will manually define it.
-                    self.shards_num = 1; // shard(shard-0)
+                    self.shard = 1; // shard(shard-0)
```
Comment (bingyanglin) [inline src/node/supervisor.rs]: I prefer to use shard_count here. For the naming of field, I prefer to use [struct name]_count for the counting, and use id2[struct name] for the hash map of ID to struct. How do you think?

File: src/stage/supervisor.rs [hunk 2]
```diff
@@ -71,51 +67,51 @@ impl SupervisorBuilder {
...
+                .reporter(reporter_num)
+                .session(self.session)
+                .streams(streams)
+                .shard(self.shards.clone())
+                .address(self.address.clone())
                 .tx(reporter_tx)
                 .rx(reporter_rx)
-                .shard(self.shard.clone())
-                .address(self.address.clone());
+                .supervisor_tx(self.tx.clone())
```
Comment (bingyanglin) [inline src/stage/supervisor.rs]: For the naming rules I prefer to use node_tx here for ease of reading (and precise identification), so as other supervisor_tx field in other structs (reporter/stage/node/etc.) How do you think?

File: src/cluster/supervisor.rs [hunk 3]
```diff
@@ -20,27 +19,27 @@ enum Event {
...
+    reporters: u8,
     // pub supervisor_tx:
 }
 
 impl SupervisorBuilder {
     pub fn new() -> Self {
         SupervisorBuilder {
             address: None,
-            reporters_num: None,
+            reporters: 1,
```
[the remark this answers is not in the document]
Comment (wusyong) [reply src/cluster/supervisor.rs]: Use ditto next time.

File: src/cluster/supervisor.rs [hunk 4]
```diff
@@ -50,7 +49,7 @@ impl SupervisorBuilder {
 // suerpvisor state struct
 pub struct Supervisor {
     address: Address,
-    reporters_num: ReporterNum,
+    reporters: u8,
     tx: Sender,
```
[the remark this answers is not in the document]
Comment (bingyanglin) [reply src/cluster/supervisor.rs]: Modifed

File: src/cluster/supervisor.rs [hunk 5]
```diff
@@ -50,7 +49,7 @@ impl SupervisorBuilder {
 // suerpvisor state struct
 pub struct Supervisor {
     address: Address,
-    reporters_num: ReporterNum,
+    reporters: u8,
```
[the remark this answers is not in the document]
Comment (bingyanglin) [reply src/cluster/supervisor.rs]: Modified

File: src/node/supervisor.rs [hunk 6]
```diff
@@ -23,50 +20,52 @@ pub enum Event {
...
 
 // suerpvisor state struct
 pub struct Supervisor {
-    address: Address,
-    reporters_num: ReporterNum,
+    address: String,
+    reporters: u8,
     spawned: bool,
-    tx: Sender,
+    pub tx: Sender,
```
[the remark this answers is not in the document]
Comment (bingyanglin) [reply src/node/supervisor.rs]: This is just for test purpose so just leave it currently and add a TODO comment.

File: src/cluster/supervisor.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (LouayKamel) [reply src/cluster/supervisor.rs]: soon will back to work on cluster supervisor

**Pull request merged**